trace-utils/src/trace_utils.rs

```diff
@@ -40,6 +45,210 @@ pub async fn get_traces_from_request_body(body: Body) -> anyhow::Result<(usize,
+        anyhow::bail!("Expected an array of exactly 12 elements in a span, got {span_size}");
+    }
+    //0 - service
+    match read_value(reader)? {
+        Value::Integer(s) => {
+            span.service = str_from_dict(dict, s)?;
+        },
+        val => anyhow::bail!("Error reading span service, value is not an integer and can't be looked up in dict: {val}")
+    };
+    // 1 - name
+    match read_value(reader)? {
+        Value::Integer(s) => {
+            span.name = str_from_dict(dict, s)?;
+        },
+        val => anyhow::bail!("Error reading span name, value is not an integer and can't be looked up in dict: {val}")
+    };
+    // 2 - resource
+    match read_value(reader)? {
+        Value::Integer(s) => {
+            span.resource = str_from_dict(dict, s)?;
+        },
+        val => anyhow::bail!("Error reading span resource, value is not an integer and can't be looked up in dict: {val}")
+    };
+    // 3 - trace_id
+    match read_value(reader)? {
+        Value::Integer(i) => {
+            span.trace_id = i.as_u64().ok_or_else(|| {
+                anyhow!("Error reading span trace_id, value is not an integer: {i}")
+            })?;
+        }
+        val => anyhow::bail!("Error reading span trace_id, value is not an integer: {val}"),
+    };
+    // 4 - span_id
+    match read_value(reader)? {
+        Value::Integer(i) => {
+            span.span_id = i.as_u64().ok_or_else(|| {
+                anyhow!("Error reading span span_id, value is not an integer: {i}")
+            })?;
+        }
+        val => anyhow::bail!("Error reading span span_id, value is not an integer: {val}"),
+    };
+    // 5 - parent_id
+    match read_value(reader)? {
+        Value::Integer(i) => {
+            span.parent_id = i.as_u64().ok_or_else(|| {
+                anyhow!("Error reading span parent_id, value is not an integer: {i}")
+            })?;
+        }
+        val => anyhow::bail!("Error reading span parent_id, value is not an integer: {val}"),
+    };
+    //6 - start
+    match read_value(reader)? {
+        Value::Integer(i) => {
+            span.start = i
+                .as_i64()
+                .ok_or_else(|| anyhow!("Error reading span start, value is not an integer: {i}"))?;
+        }
+        val => anyhow::bail!("Error reading span start, value is not an integer: {val}"),
+    };
+    //7 - duration
+    match read_value(reader)? {
+        Value::Integer(i) => {
+            span.duration = i.as_i64().ok_or_else(|| {
+                anyhow!("Error reading span duration, value is not an integer: {i}")
+            })?;
+        }
+        val => anyhow::bail!("Error reading span duration, value is not an integer: {val}"),
+    };
+    //8 - error
+    match read_value(reader)? {
+        Value::Integer(i) => {
+            span.error = i
+                .as_i64()
+                .ok_or_else(|| anyhow!("Error reading span error, value is not an integer: {i}"))?
+                as i32;
```

Comment:
Is the `as i32` expected here in error? If so, should we add a comment on why we do this here and not for the other ones?

Reply:
Weird to read on github, but yeah an i32 is required. I don't think a comment is necessary because the compiler complains with out it.

Here's the i32: https://github.com/DataDog/libdatadog/blob/main/trace-protobuf/src/pb/span.proto#L49